internal/worker/caasunitterminationworker/worker.go

```diff
@@ -80,7 +81,7 @@ func (w *terminationWorker) loop(c <-chan os.Signal) (err error) {
 	select {
 	case <-c:
 		w.logger.Infof("terminating due to SIGTERM")
-		term, err := w.state.UnitTerminating(w.agent.CurrentConfig().Tag().(names.UnitTag))
+		term, err := w.state.UnitTerminating(context.TODO(), w.agent.CurrentConfig().Tag().(names.UnitTag))
```

Comment:
Add scopedContext function to the worker, then we can get rid of the TODO.

Reply:
Yeah, is mentioned in the cover description as a todo for a followup. This PR is already way too big.